### PR TITLE
[Metricbeat](Etcd-Leader)Followers wont report leader metrics

### PR DESCRIPTION
**Pull request by odacremolbap**
Related to https://github.com/elastic/beats/pull/10591

Etcd Leader metrics return HTTP status 403 when pulling from a follower. This is pushing error events to the configured output.

We don't know beforehand if a node is leader or follower, this PR assumes we need to ask each node in the cluster and when a 403 code is returned containing the JSON response `{"message":"not current leader"}`, we skip sending an event with an error message.

The returned body condition needs to be checked to distinguish this case from other 403 code scenarios (like wrong credentials)

### CODE REVIEW

File: metricbeat/module/etcd/leader/leader.go [hunk 1]
```diff
@@ -71,14 +80,46 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
+	if err = json.Unmarshal(content, &jsonResponse); err == nil {
+		if retMessage := jsonResponse[msgElement]; retMessage != "" {
+			// there is an error message element, let's use it
+
+			// If a 403 is returned and {"message":"not current leader"}
+			// do not consider this an error
+			// do not report events since this is not a leader
+			if res.StatusCode == http.StatusForbidden &&
+				retMessage == msgValueNonLeader {
+				return nil
```
Comment (exekias) [inline metricbeat/module/etcd/leader/leader.go]: maybe add a debug message?

Comment (kaiyan-sheng) [reply metricbeat/module/etcd/leader/leader.go]: Agree on a debug/info level log here 👍

Comment (odacremolbap) [reply metricbeat/module/etcd/leader/leader.go]: Cool, thanks for the review.

I'm ok with all debug info, but let me clarify. An etcd cluster is usually 3 or 5 instances, only one of those instances will show an event, not the other 2, we can consider that _if_ statement above the _normal_ path.
Not an error, not an unexpected path.

I'll add a **Debug** log in there if that's ok with you.

Comment (exekias) [reply metricbeat/module/etcd/leader/leader.go]: I agree it's not an error, still some users may not understand why they are not getting metrics from a node where this module is running :+1:, debug log will help them understand that

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -156,6 +156,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add Filebeat envoyproxy module. {pull}11700[11700]
 - Add apache2(httpd) log path (`/var/log/httpd`) to make apache2 module work out of the box on Redhat-family OSes. {issue}11887[11887] {pull}11888[11888]
 - Add support to new MongoDB additional diagnostic information {pull}11952[11952]
+-
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: one extra line here 😄

**Comment by odacremolbap** [comment]
jenkins, test this please

**Pull request merged**